Modules/POSIX/Sources/POSIX/Regex.swift

```diff
@@ -0,0 +1,319 @@
+#if os(Linux)
+    @_exported import Glibc
+#else
+    @_exported import Darwin.C
+#endif
+
+import Foundation
+
+
+public struct RegexError : Error {
+    let description: String
+
+    static func error(from result: Int32, preg: inout regex_t) -> RegexError {
+        var buffer = [Int8](repeating: 0, count: Int(BUFSIZ))
+        regerror(result, &preg, &buffer, buffer.count)
+        let description = String(cString: buffer)
+        return RegexError(description: description)
+    }
+}
+
+
+public final class Regex {
+
+    public struct Options : OptionSet {
+        public let rawValue: Int32
+
+        public init(rawValue: Int32) {
+            self.rawValue = rawValue
+        }
+
+        public static let basic            = Options(rawValue: 0)
+        public static let extended         = Options(rawValue: 1)
+        public static let caseInsensitive  = Options(rawValue: 2)
+        public static let newLineSensitive = Options(rawValue: 4)
+        public static let resultOnly       = Options(rawValue: 8)
+    }
+
+    public struct MatchOptions : OptionSet {
+        public let rawValue: Int32
+
+        public init(rawValue: Int32) {
+            self.rawValue = rawValue
+        }
+
+        public static let firstCharacterNotAtBeginningOfLine = MatchOptions(rawValue: REG_NOTBOL)
+        public static let lastCharacterNotAtEndOfLine        = MatchOptions(rawValue: REG_NOTEOL)
+    }
+
+
+    var preg = regex_t()
+
+    /// Constructs a Regex using the given string pattern and `RegexOptions`.
+    ///
+    /// - parameter pattern: string holding the Regex pattern.
+    /// - parameter options: Options such as `Regex.Options.basic` (BRE) or `Regex.Options.extended` (ERE).
+    ///
+    /// - throws: `RegexError` if the given pattern is not a valid Regex.
+    public init(pattern: String, options: Options = .extended) throws {
```

Comment:
I think we can remove the `pattern` label. It's pretty obvious the `String` is the pattern.